### PR TITLE
Add new rule `no-scope-outside-table-headings`

### LINKED ISSUE #2412: Proposal to add new a11y rule restricting the use of the scope attribute to `th` elements
Currently, ember-template-lint does not disallow HTML elements other than the <th> element from using the scope attribute. Scope is used on `<th>` elements to clarify the relationship between a table's header cells and data cells for screen readers. Because placing this attribute on other HTML elements has no effect, it is a best practice that only `<th>` elements declare a value for scope.

This proposes the authoring of a new rule that enforces that the scope attribute is only used on `<th>` elements. This is essentially a port of JSXs A11y [scope rule](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/scope.md).

References: 
- [H63: Using the scope attribute to associate header cells and data cells in data tables](https://www.w3.org/TR/WCAG20-TECHS/H63.html)
- [Tables Concepts](https://www.w3.org/WAI/tutorials/tables/)

### PR DESCRIPTION
**Pull request by judithhinlung**
This PR adds the new rule `scope` which fixes #2412 .
 
### Background
The scope attribute is used on `<th>` elements to clarify the relationship between a table's header cells and data cells for screen readers. Because placing this attribute on other HTML elements has no effect, only `<th>` elements should declare a value for scope.

This new rule enforces that the scope attribute is only used on `<th>` elements.  

References: 
- [H63: Using the scope attribute to associate header cells and data cells in data tables](https://www.w3.org/TR/WCAG20-TECHS/H63.html)
- [Tables Concepts](https://www.w3.org/WAI/tutorials/tables/)

 **Allowed**:
 
 ```
# examples
<th scope="row">some header cell</th>
<CustomHeader scope={{foo}} />
```

**Forbidden**:

```
# examples
<a scope="col"></a>
<table scope="rowgroup"></table>
```

### Testing
```
Test Suites: 130 passed, 130 total
Tests:       5 skipped, 7318 passed, 7323 total
Snapshots:   945 passed, 945 total
Time:        125.954 s
Ran all test suites.
✨  Done in 133.92s.

```

### CODE REVIEW

**Comment by scalvert** [comment]
Hey @judithhinlung, I know this is still in progress, but can we rename the rule to something a bit less general? Maybe something like `require-valid-scope` or something?

File: test/unit/rules/no-scope-outside-table-headings-test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    '<th scope="row">Some table heading></th>',
+    `
+<table>
+<th scope="col">Table header</th>
+<td>Some data</td>
+</table>
+`,
+    '<CustomComponent scope />',
+    '<CustomComponent scope={{foo}} />',
+  ],
```
Comment (bmish) [inline test/unit/rules/no-scope-outside-table-headings-test.js:18]: Can we add these to `good`:
* `{{foo-component scope="row"}}`
* `<CustomComponent scope="row" />`

File: test/unit/rules/no-scope-outside-table-headings-test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  bad: [
```
Comment (bmish) [inline test/unit/rules/no-scope-outside-table-headings-test.js:19]: Can we add this to bad just to show that we don't care what the value is?
* `<td scope>`

**Review by bmish — CHANGES_REQUESTED** [review]

File: docs/rule/no-scope-outside-table-headings.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+```
+
+This rule **allows** the following:
+
+```hbs
+<th scope="row">A header cell</th>
+<CustomHeader scope={{foo}} />
+```
+
+## References
```
Comment (bmish) [inline docs/rule/no-scope-outside-table-headings.md:23]: Can you also link to something specific about `scope`? https://www.w3schools.com/tags/att_th_scope.asp

**Review by bmish — APPROVED** [review]

File: docs/rule/no-scope-outside-table-headings.md (hunk 3 above)
Comment (bmish) [inline docs/rule/no-scope-outside-table-headings.md:23]: Can you also link to 

* https://www.w3.org/TR/WCAG20-TECHS/H63.html
* And the corresponding React lint rule

Comment (bmish) [reply docs/rule/no-scope-outside-table-headings.md:23]: Can you also link to 

* https://www.w3.org/TR/WCAG20-TECHS/H63.html
* And the JSXs A11y [scope rule](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/scope.md).

**Comment by judithhinlung** [comment]
Hi @bmish, is it okay to merge this pr?

**Pull request merged**